Codes/Car/LaneDetection_youngsik.cpp

```diff
@@ -0,0 +1,390 @@
+                    double minContourArea = 10.0; // ������ ������ ����� ���ֱ� ���� �Ӱ谪
+
+                    if (area >= minContourArea)
+                    {
+                        /*
+                            Moments�� ������(������)�� Ư¡�� ���
+                            mu.m00: 0�� ���Ʈ, ������ ���� (���� ��)
+                            mu.m10: 1�� ���Ʈ X(X ��ǥ�� ���� ��)
+                            mu.m01 : 1�� ���Ʈ Y(Y ��ǥ�� ���� ��)
+                            mu.m20 : 2�� ���Ʈ X(X ��ǥ�� ������ ���� ��)
+                            mu.m02 : 2�� ���Ʈ Y(Y ��ǥ�� ������ ���� ��)
+                            mu.m11 : 2�� ���Ʈ XY(X�� Y ��ǥ�� ���� ���� ��)
+                        */
+                        Moments mu = moments(contour);
+
+                        // �߽� ��ǥ ���
+                        Point2f centroid(mu.m10 / mu.m00, mu.m01 / mu.m00);
+
+                        // contours �߽� ��ǥ�� �� �׸���
+                        Point2f centcircle(centroid.x + rect.x, centroid.y + rect.y);
+                        centroids_left.push_back(centcircle);
+                        //circle(birdview, centcircle, 4, Scalar(0, 255, 0), -1);
+                    }
+                }
+            }
+        }
+
+        /*
+        �� ���簢�� ����(ROI)���� �߽���ǥ�� �̾ �� �������� �׸���
+
+        Mat roiLine_left = Mat::zeros(img_topview.size(), CV_8UC3);
+        for (size_t i = 0; i < centroids_left.size() - 1; i++)
+        {
+            line(roiLine_left, centroids_left[i], centroids_left[i + 1], Scalar(255, 0, 0), 7);
+        }
+        */
+        // ���簢�� ��ġ�� �����ϴ� ���� �ʱ�ȭ
+        // ������ ���� ���� ����
+        vector<Rect> rectangles_right;
+        for (int i = 0; i < numRect; i++)
+        {
+            // ���簢�� ��ǥ ���
+            int y1 = i * rectHeight;
+            int y2 = (i + 1) * rectHeight;
+            int x = 140;
+
+            Rect rect(x, y1, 70, y2 - y1);
+
+            rectangles_right.push_back(rect);
+        }
+
+        //rectangles_right �迭�� ���� rect�� �������
+        for (const auto& rect : rectangles_right)
+        {
+            rectangle(Line_img, rect, Scalar(255, 0, 0), 3); 
+        }
+
+
+        // �߽� ��ǥ�� ������ ���� �ʱ�ȭ
+        vector<Point2f> centroids_right;
+
+        for (const Rect& rect : rectangles_right)
+        {
+
+            Mat roi = LineContoure(rect);
+            vector<vector<Point>> contours;
+
+            // rectangles_left  ���� �簢������ ������ ����
+            findContours(roi, contours, RETR_EXTERNAL, CHAIN_APPROX_SIMPLE);
+
+            /*
+                ROI �������� ����� contours ó��
+                rectangles_left  ���� �簢������ ó���ϱ� ������ contours �Ѱ��� ����
+            */
+
+            // 1) ���� �������� ������ ���� ���
+            if (contours.empty())
+            {
+                //������ ���簢���� �߽����� ���Ƿ� ����
+                Point2f centcircle(rect.x + rect.width / 2, rect.y + rect.height / 2);
```

Comment:
연산 속도를 빠르게 하기 위해 `/2` 대신 `>>1` 연산으로 대체하는 건 어떨까요?

```suggestion
                Point2f centcircle(rect.x + rect.width >> 2, rect.y + rect.height >> 2);
```